test/e2e/app/state.go

```diff
@@ -171,14 +172,18 @@ func (s *State) Rollback() error {
 }
 
 // hashItems hashes a set of key/value items.
-func hashItems(items map[string]string) []byte {
+func hashItems(items map[string]string, height uint64) []byte {
 	keys := make([]string, 0, len(items))
 	for key := range items {
 		keys = append(keys, key)
 	}
 	sort.Strings(keys)
 
 	hasher := sha256.New()
+	b := make([]byte, 8)
+	binary.BigEndian.PutUint64(b, height)
+	_, _ = hasher.Write(b)
+	_, _ = hasher.Write([]byte{0})
```

Comment:
```suggestion
	var b [8]byte
	binary.BigEndian.PutUint64(b[:], height)
	_, _ = hasher.Write(b[:])
```

The encoding of the value is fixed-length, so we don't need to pad anything.